src/VideoReceiver/GstVideoReceiver.h

```diff
@@ -158,18 +159,14 @@ protected slots:
 
     unsigned            _timeout;
 
-    QWaitCondition      _taskQueueUpdate;
-    QMutex              _taskQueueSync;
-    QQueue<Task>        _taskQueue;
-    bool                _shutdown;
+    Worker              _apiHandler;
+    Worker              _notificationHandler;
 
-    static const char*  _kFileMux[FILE_FORMAT_MAX - FILE_FORMAT_MIN];
-#else
-private:
-#endif
+    bool                _endOfStream;
 
-    std::atomic<bool>   _streaming;
-    std::atomic<bool>   _decoding;
-    std::atomic<bool>   _recording;
-    std::atomic<quint32>_videoSize;
+    static const char*  _kFileMux[FILE_FORMAT_MAX - FILE_FORMAT_MIN];
 };
+
+void* createVideoSink(void* widget);
```

Comment:
Both `createVideoSink()` and `initializeVideoReceiver` should be static methods of some (root) video streaming class instead of some global function. That's primarily to keep it all together and within the proper scope (namespace).